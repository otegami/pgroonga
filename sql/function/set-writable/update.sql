CREATE TABLE memos (
  content text
);

CREATE INDEX pgrn_index ON memos USING PGroonga (content);

INSERT INTO memos VALUES ('Groonga is fast!');

SELECT pgroonga_set_writable(false);

UPDATE memos SET content = 'PGroonga is fast!';

SELECT * FROM memos;
SELECT pgroonga_command('select',
			ARRAY[
			  'table', pgroonga_table_name('pgrn_index'),
			  'output_columns', '_id, content'
			])::jsonb->>1;

SELECT pgroonga_set_writable(true);

UPDATE memos SET content = 'PGroonga is fast!';

SELECT * FROM memos;
SELECT pgroonga_command('select',
			ARRAY[
			  'table', pgroonga_table_name('pgrn_index'),
			  'output_columns', '_id, content'
			])::jsonb->>1;

DROP TABLE memos;
