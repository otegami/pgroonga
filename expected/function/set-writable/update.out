CREATE TABLE memos (
  content text
);
CREATE INDEX pgrn_index ON memos USING PGroonga (content);
INSERT INTO memos VALUES ('Groonga is fast!');
SELECT pgroonga_set_writable(false);
 pgroonga_set_writable 
-----------------------
 t
(1 row)

UPDATE memos SET content = 'PGroonga is fast!';
ERROR:  pgroonga: [insert] can't insert a record while pgroonga.writable is false
SELECT * FROM memos;
     content      
------------------
 Groonga is fast!
(1 row)

SELECT pgroonga_command('select',
			ARRAY[
			  'table', pgroonga_table_name('pgrn_index'),
			  'output_columns', '_id, content'
			])::jsonb->>1;
                                    ?column?                                    
--------------------------------------------------------------------------------
 [[[1], [["_id", "UInt32"], ["content", "LongText"]], [1, "Groonga is fast!"]]]
(1 row)

SELECT pgroonga_set_writable(true);
 pgroonga_set_writable 
-----------------------
 f
(1 row)

UPDATE memos SET content = 'PGroonga is fast!';
SELECT * FROM memos;
      content      
-------------------
 PGroonga is fast!
(1 row)

SELECT pgroonga_command('select',
			ARRAY[
			  'table', pgroonga_table_name('pgrn_index'),
			  'output_columns', '_id, content'
			])::jsonb->>1;
                                                 ?column?                                                 
----------------------------------------------------------------------------------------------------------
 [[[2], [["_id", "UInt32"], ["content", "LongText"]], [1, "Groonga is fast!"], [2, "PGroonga is fast!"]]]
(1 row)

DROP TABLE memos;
